src/test/deploy/functions/release/sourceTokenScraper.spec.ts

```diff
@@ -31,14 +31,63 @@ describe("SourcTokenScraper", () => {
   it("provides tokens from an operation", async () => {
     const scraper = new SourceTokenScraper();
     // First result comes right away
-    await expect(scraper.tokenPromise()).to.eventually.be.undefined;
+    await expect(scraper.getToken()).to.eventually.be.undefined;
 
     scraper.poller({
       metadata: {
         sourceToken: "magic token",
         target: "projects/p/locations/l/functions/f",
       },
     });
-    await expect(scraper.tokenPromise()).to.eventually.equal("magic token");
+    await expect(scraper.getToken()).to.eventually.equal("magic token");
+  });
+
+  it("refreshes token after timer expires", async () => {
+    const scraper = new SourceTokenScraper(10);
+    await expect(scraper.getToken()).to.eventually.be.undefined;
+    scraper.poller({
+      metadata: {
+        sourceToken: "magic token",
+        target: "projects/p/locations/l/functions/f",
+      },
+    });
+    await expect(scraper.getToken()).to.eventually.equal("magic token");
+    const timeout = (duration: number): Promise<void> => {
+      return new Promise<void>((resolve) => setTimeout(resolve, duration));
+    };
+    await timeout(50);
```

Comment:
In the future, try to instrument code without sleeping in your test. You can inject a clock for example.